packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -1,79 +1,49 @@
 import { MutationTypes, CozyLink, getDoctypeFromOperation } from 'cozy-client'
 import PouchDB from 'pouchdb'
 import PouchDBFind from 'pouchdb-find'
-import fromPairs from 'lodash/fromPairs'
 import omit from 'lodash/omit'
+import defaults from 'lodash/defaults'
 
 import { withoutDesignDocuments } from './helpers'
 import { getIndexNameFromFields, getIndexFields } from './mango'
-
+import * as jsonapi from './jsonapi'
+import PouchManager from './PouchManager'
 PouchDB.plugin(PouchDBFind)
 
-const pipe = fn => res => {
-  try {
-    fn(res)
-  } catch (e) {
-    console.warn('Error during pipe', e)
-  }
-  return res
-}
+const sanitized = doc => omit(doc, '_type')
 
-const ensureHasBothIds = obj => {
-  if (!obj._id && obj.id) {
-    obj._id = obj.id
-  }
-  if (!obj._id && obj.id) {
-    obj.id = obj._id
-  }
-  return obj
+const parseMutationResult = (original, res) => {
+  return { ...original, ...omit(res, 'ok') }
 }
 
-const addDoctype = (arr, doctype) => arr.map(x => ({ ...x, _type: doctype }))
+const doNothing = () => {}
 
-const pouchResToJSONAPI = (res, isArray, doctype) => {
-  if (isArray) {
-    const docs = res.rows || res.docs
-    const offset = res.offset || 0
-    return {
-      data: addDoctype(docs, doctype).map(ensureHasBothIds),
-      meta: { count: docs.length },
-      skip: offset,
-      next: offset + docs.length <= res.total_rows
-    }
-  } else {
-    return {
-      data: ensureHasBothIds(res)
-    }
-  }
+const DEFAULT_OPTIONS = {
+  replicationInterval: 30 * 1000
 }
 
-const createPouches = doctypes => {
-  return fromPairs(
-    doctypes.map(doctype => [
-      doctype,
-      {
-        db: new PouchDB(doctype)
-      }
-    ])
-  )
+const addBasicAuth = (url, basicAuth) => {
+  return url.replace('//', `//${basicAuth}`)
 }
 
-const sanitized = doc => omit(doc, '_type')
-
-const parseMutationResult = (original, res) => {
-  return { ...original, ...omit(res, 'ok') }
+export const getReplicationURL = (uri, token, doctype) => {
+  const basicAuth = token.toBasicAuth()
+  const authenticatedURL = addBasicAuth(uri, basicAuth)
+  return `${authenticatedURL}/data/${doctype}`
 }
 
-const doNothing = () => {}
+export const isExpiredTokenError = pouchError => {
+  return pouchError.error == 'code=400, message=Expired token'
```

Comment:
It is recommended to never use the evil twins, aka `==` and `!=` by Crockford.
[More details on StackOverflow](https://stackoverflow.com/a/359509/178575).

Reply:
Thanks !